modules/aggregation-layers/src/screen-grid-layer/screen-grid-layer-vertex.glsl.ts

```diff
@@ -23,63 +23,35 @@ export default `\
 #define SHADER_NAME screen-grid-layer-vertex-shader
 #define RANGE_COUNT 6
 
-in vec3 positions;
-in vec3 instancePositions;
-in vec4 instanceCounts;
+in vec2 positions;
+in vec2 instancePositions;
+in float instanceWeights;
 in vec3 instancePickingColors;
 
 uniform float opacity;
-uniform vec3 cellScale;
-uniform vec4 minColor;
-uniform vec4 maxColor;
-uniform vec4 colorRange[RANGE_COUNT];
+uniform ivec2 binCount;
+uniform vec2 gridSizeClipspace;
+uniform vec2 cellSizeClipspace;
 uniform vec2 colorDomain;
-uniform bool shouldUseMinMax;
-uniform sampler2D maxTexture;
+uniform sampler2D colorRange;
 
 out vec4 vColor;
-out float vSampleCount;
-
-vec4 quantizeScale(vec2 domain, vec4 range[RANGE_COUNT], float value) {
-  vec4 outColor = vec4(0., 0., 0., 0.);
-  if (value >= domain.x && value <= domain.y) {
-    float domainRange = domain.y - domain.x;
-    if (domainRange <= 0.) {
-      outColor = colorRange[0];
-    } else {
-      float rangeCount = float(RANGE_COUNT);
-      float rangeStep = domainRange / rangeCount;
-      float idx = floor((value - domain.x) / rangeStep);
-      idx = clamp(idx, 0., rangeCount - 1.);
-      int intIdx = int(idx);
-      outColor = colorRange[intIdx];
-    }
-  }
-  outColor = outColor / 255.;
-  return outColor;
-}
+out float vIsValid;
```

Comment:
Time to start using integer values in shaders since we are now always on GLSL 3.00? 
```
out int vIsValid;
```